src/main/java/net/marcellperger/mathexpr/parser/Parser.java

```diff
@@ -81,11 +80,13 @@ public MathSymbol parseInfixPrecedenceLevel(int level) throws ExprParseException
         if(level == 0) {
             return parseParensOrLiteral();
         }
-
+        // TODO remove this next-line:
+        if(SymbolInfo.PREC_TO_INFO_MAP.get(level) == null) { return parseInfixPrecedenceLevel(level - 1); }
         Set<SymbolInfo> symbols = Util.requireNonEmptyNonNull(SymbolInfo.PREC_TO_INFO_MAP.get(level));
         @Nullable GroupingDirection dirn = symbols.stream()
             .map(sm -> sm.groupingDirection).distinct().collect(UtilCollectors.singleItem());
-        assert dirn == GroupingDirection.LeftToRight: "RTL/unknown operators not implemented yet";
+        // if(dirn != GroupingDirection.LeftToRight) { return parseInfixPrecedenceLevel(level - 1); }  // TODO TDD: remove
+        // assert dirn == GroupingDirection.LeftToRight: "RTL/unknown operators not implemented yet"; // TODO
         Map<String, SymbolInfo> infixToSymbolInfo = symbols.stream().collect(  // TODO pre-compute/cache these
```

Comment:
These should somehow be cached/precomputed - not sure if it should be this PR or later?